sale_delivery_state/tests/test_delivery_state.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2018 Akretion (http://www.akretion.com).
+# @author Benoît GUILLOT <benoit.guillot@akretion.com>
+# Copyright 2018 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests import SavepointCase
+
+
+class TestDeliveryState(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.order = cls.env.ref("sale_delivery_state.sale_order_1")
+        cls.delivery_cost = cls.env["product.product"].create(
+            {"name": "delivery", "type": "service"}
+        )
+
+    def test_no_delivery(self):
+        self.assertEqual(self.order.delivery_state, "no")
+
+    def test_unprocessed_delivery(self):
+        self.order.action_confirm()
+        self.assertEqual(self.order.delivery_state, "unprocessed")
+
+    def test_partially(self):
+        self.order.action_confirm()
+        self.order.order_line[0].qty_delivered = 2
+        self.assertEqual(self.order.delivery_state, "partially")
+
+    def test_delivery_done(self):
+        self.order.action_confirm()
+        for line in self.order.order_line:
+            line.qty_delivered = line.product_uom_qty
+        self.assertEqual(self.order.delivery_state, "done")
+
+    def mock_delivery(self):
```

Comment:
I didn't notice this before...  why don't you use `mock`? 
If you need to patch a filed value you can use `PropertyMock`
Check this example https://github.com/OCA/wms/blob/63620cdd6abf14198c11967d4660c0e421f3a877/shopfloor/tests/test_checkout_scan_package_action.py#L392

Reply:
please cherry-pick my commit here https://github.com/OCA/sale-workflow/pull/1649/commits/1464c53fae5ab43b04136803ae9bf5e86254a40f